client/authority-discovery/src/worker/tests.rs

```diff
@@ -385,62 +382,35 @@ fn terminate_when_event_stream_terminates() {
 	});
 
 	let (_to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
+	let worker = Worker::new(
 		from_service,
 		test_api,
 		network.clone(),
 		vec![],
 		dht_event_rx.boxed(),
-		Role::Authority(key_store),
+		Role::Authority(key_store.into()),
 		None,
 	);
 
-	block_on(async {
-		assert_eq!(Poll::Pending, poll!(&mut worker));
-
+	let timer = async {
+		Delay::new(Duration::from_secs(1)).await;
```

Comment:
I think we should try hard not to depend on any timing assumptions in unit tests. It is likely fine here, but say the whole process is preempted for a second, ... . Why not do as we did before, but instead of `poll!(&mut worker)` you would need to `poll!` the future returned by `worker.run()`.